### ISSUE TITLE
fatal error: incompatible librhash version is loaded: 1.3.9

### ISSUE THREAD

**Issue by mariomadproductions** [rhash/RHash#141]
When I build rhash v1.4.1 (in Linux Mint 20.1) and then run it, it says "fatal error: incompatible librhash version is loaded: 1.3.9". I did have v1.3.9 installed in the repos before, but I uninstalled it due to it sefaulting when trying to recursively verify files (issue #104).

I sort-of solved this by installing the librhash .deb from here https://launchpad.net/ubuntu/+source/rhash/1.4.1-1/+build/20788722. But this probably isn't the right way to go about things.

**Comment by rhash** [comment]
You uninstalled rhash package but left the librhash0 package. Newly compiled binary loaded the installed shared library librhash.so v1.3.9 and reported version incompatibility.

When you installed librhash*.deb from Ubuntu, dpkg removed your old librhash0_1.3.9-1, so it solved the issue.

The correct way was to build and install rhash DEB package of the required version. Then dpkg/apt would require installation of the correct librhash version by dependency.

**Comment by mariomadproductions** [comment]
Okay. I removed rhash from `/usr/local/bin/` and uninstalled the librhash0 package (I didn't do the latter in my last, failed, attempt). Then with the v1.4.1 source code I did
```
sudo ./configure --enable-openssl-runtime
sudo make
sudo make install
```
This apparently installs rhash and the correct librhash version. I guess you are saying if I packaged the build into a deb file, then dpkg would get the librhash0 from the Ubuntu unstable repo (even if Linux Mint 20.1 doesn't use the unstable repo)?
Anyway, I guess this is basically solved now, thanks.
